DIPs/DIP1002.md

```diff
@@ -131,6 +198,114 @@ finally
 }
 ```
 
+Note that a `scope(success)` placed in the `try` statement cannot be used to
+accurately reproduce this behavior because exceptions thrown by
+`depends_on_success_of_thing` would in that case be caught by `catch` statements
+intended to handle errors produced by `do_a_thing`, but not those produced by
+`depends_on_success_of_thing`.
+
+#### Code in Phobos that could be rewritten using `else`
+
+The best examples of code that can be more elegantly written using this
+`else` clause occur in application logic, such as the example described
+above where a values were read from sensors and saved to a server. However,
+there are a few places in Phobos that could be more readable and not
+require success flags if this new syntax were employed.
+
+Here is a unittest in `phobos/std/range/package.d`:
+
+``` D
+unittest
+{
+    bool ok = true;
+    try
+    {
+        auto r2 = assumeSorted([ 677, 345, 34, 7, 5 ]);
+        debug ok = false;
+    }
+    catch (Throwable)
+    {
+    }
+    assert(ok);
+}
+```
+
+With this addition to exception handling, it could be rewritten like so:
+
+``` D
+unittest
+{
+    try
+    {
+        auto r2 = assumeSorted([ 677, 345, 34, 7, 5 ]);
+    }
+    catch (Throwable)
+    {
+    }
+    else
+    {
+        debug assert(0);
+    }
+}
+```
+
+Here is a portion of a unittest in `phobos/std/variant.d`:
+
+``` D
+// try failing conversions
+bool failed = false;
+try
+{
+    auto d = c.get!(int);
+}
+catch (Exception e)
+{
+    //writeln(stderr, e.toString);
+    failed = true;
+}
+assert(failed); // :o)
+```
+
+It could be rewritten like so:
+
+``` D
+// try failing conversions
+try
+{
+    auto d = c.get!(int);
+}
+catch (Exception e)
+{   
+}
+else
+{
+    assert(0); // :o)
```

Comment:
Please note that retaining the comment is very important, should this code be refactored